File: MoneyFlowAPI/Application/DTOs/DTO_Dashboard.cs
namespace MoneyFlowAPI.Application.DTOs
{
    public class DTO_Dashboard
    {
        public string MonthLabel { get; set; } = string.Empty;

        public decimal CurrentBalance { get; set; }

        public decimal MonthlyIncome { get; set; }

        public decimal MonthlyExpense { get; set; }

        public decimal NetSavings { get; set; }

        public int MonthlyTransactionsCount { get; set; }

        public decimal SavingsRate { get; set; }

        public string LatestTransactionDescription { get; set; } = "Sem movimentos";

        public DateOnly? LatestTransactionDate { get; set; }

        public List<DTO_DashboardTopCategory> TopCategories { get; set; } = new();
    }
}

File: MoneyFlowUI/src/app/shared/dtos/DTO_Dashboard.ts
import { DTO_DashboardTopCategory } from './DTO_DashboardTopCategory';

export class DTO_Dashboard {
    monthLabel = '';
    currentBalance = 0;
    monthlyIncome = 0;
    monthlyExpense = 0;
    netSavings = 0;
    monthlyTransactionsCount = 0;
    savingsRate = 0;
    latestTransactionDescription = 'Sem movimentos';
    latestTransactionDate: string | null = null;
    topCategories: DTO_DashboardTopCategory[] = [];
}
